include/elements/2d/triangle_lme_element.h

```diff
@@ -0,0 +1,178 @@
+#ifndef MPM_LME_ELEMENT_H
+#define MPM_LME_ELEMENT_H
+
+#include "triangle_element.h"
+
+namespace mpm {
+
+//! Quadrilateral Local Maximum Entropy element class derived from Quadrilateral
+//! \brief Quadrilateral LME element
+//! \details quadrilateral LME element with adaptive number of nodes \n
+//! Type of elements and sizes of shape function, gradient shape function,
+//! B-matrix, indices are according to the number of nodes \n <pre>
+//!
+//! Type 1: Trinagular Element: nnodes = 36
+//!
+//!   30---------31----------32----------33----------34----------35
+//!   |  \        |  \        |  \        |  \        |  \        |
+//!   |    \      |    \      |    \      |    \      |    \      |
+//!   |      \    |      \    |      \    |      \    |      \    |
+//!   |        \  |        \  |        \  |        \  |        \  |
+//!   24---------25----------26----------27----------28----------29
+//!   |  \        |  \        |  \        |  \        |  \        |
+//!   |    \      |    \      |    \      |    \      |    \      |
+//!   |      \    |      \    |      \    |      \    |      \    |
+//!   |        \  |        \  |        \  |        \  |        \  |
+//!   20---------21-----------3-----------2----------22----------23
+//!   |  \        |  \        |  \        |  \        |  \        |
+//!   |    \      |    \      |  particle |    \      |    \      |
+//!   |      \    |      \    |  location |      \    |      \    |
+//!   |        \  |        \  |         \ |        \  |        \  |
+//!   16---------17-----------0-----------1----------18----------19
+//!   |  \        |  \        |  \        |  \        |  \        |
+//!   |    \      |    \      |    \      |    \      |    \      |
+//!   |      \    |      \    |      \    |      \    |      \    |
+//!   |        \  |        \  |        \  |        \  |        \  |
+//!   10---------11----------12----------13----------14----------15
+//!   |  \        |  \        |  \        |  \        |  \        |
+//!   |    \      |    \      |    \      |    \      |    \      |
+//!   |      \    |      \    |      \    |      \    |      \    |
+//!   |        \  |        \  |        \  |        \  |        \  |
+//!   4-----------5-----------6-----------7-----------8-----------9
```

Comment:
Actually, I have a question: in the pattern that you draw above, there should be no difference between the shape function given by structural quad elements and the triangular right? Since the shape function is only a function of the relative position between the particle to the node.

Reply:
You are totally right !!! You will only see the difference in unstructured mesh, but it is way more difficult to draw with backslash, slash, underscore, etc jajaj